### ISSUE TITLE
Improve developer onboarding

### ISSUE THREAD

**Issue by zmbc** [mapseed/platform#686]
- Update README and make it [Standard Readme](https://github.com/RichardLitt/standard-readme/blob/master/spec.md) compliant; this will mainly benefit users but also developers via the [contribute section](https://github.com/RichardLitt/standard-readme/blob/master/spec.md#contribute)
- As much as possible, follow [Scripts To Rule Them All](https://github.com/github/scripts-to-rule-them-all) pattern
- Create developer documentation in the Wiki detailing technical architecture, including a diagram
- Move user-centered design related content from Doku wiki
- Unify or justify separation of different coordination techniques, and document separations in wiki (e.g. "we use GitHub projects for higher-level plans, and meta issues for relatively low-level plans only a bit larger in scope than a standard issue" or whatever internal logic we decide on)
- Provide an email or other contact method for requests for invites to ChatGrape

**Comment by goldpbear** [comment]
I've been thinking that another useful wiki page to have would be some kind of config glossary. Even though we'll eventually move to a config that is edited via a GUI, in the meantime it would be nice to have a page that details all the sections of the config, all the various settings that are possible, and common pitfalls to watch out for (such as duplicating layer names, not creating sidebar entries for new layers, etc.).

Plus, I think a wiki like this will come in useful later when we do attempt to convert the config to a writeable format with a nice GUI front end.

**Comment by zmbc** [comment]
@goldpbear Agreed, we should absolutely have better config documentation. That's not _quite_ in the scope of this issue (this is about developer onboarding) but still totally true. I plan to initiate a user documentation push after I complete this issue.

**Comment by LukeSwart** [comment]
in re:
>Provide an email or other contact method for requests for invites to ChatGrape

I think we should just link them here: https://gitter.im/mapseed/Lobby That's the same link we're using on our website as well, and it's probably just a matter of time before we migrate to gitter from chatgrape as well.

**Comment by zmbc** [comment]
> it's probably just a matter of time before we migrate to gitter from chatgrape as well.

I like the phrase "matter of time" because it makes it sound like a force of nature out of our control. :laughing:

I, for one, am very much in favor of Gitter and if we can get a consensus on using that I would love to help migrate us. Most likely we would want to back up our ChatGrape chat histories. If we're not going to do Gitter for ourselves, though, then let's not send people there. We don't want to have two separate chat systems.

**Comment by LukeSwart** [comment]
I talked with some folks about the ChatGrape -> Gitter migration, and I got this interesting feedback:

>There's just something nerve-racking that I can't shake when it comes to me being always online and connected to a public chat room

So this is basically a privacy issue. Perhaps there is a chat application that doesn't broadcast your online activity?

We can still use Gitter to post news, updates, and help new users, but maybe we should continue using ChatGrape until we figure this out?

Also, AFAIK, IRC doesn't broadcast your online activity. So maybe we can consider migrating to IRC instead? I believe there are some user friendly IRC bots out there...

We have an IRC channel at #mapseed, and a helpful website for folks to get going at chat.mapseed.org. It's far from the slickness of Slack or Gitter, but it's a start.

**Comment by zmbc** [comment]
@LukeSwart Gitter has private rooms, though. It says [here](https://billing.gitter.im/) that the only limitation to the free version is a limit of 25 people in a private room, but I think that should be fine for us.

To be clear, this is feedback you heard from someone working on our project, or someone who uses Gitter? If the former, they may simply not know about private rooms.

I would set one up for us but I think only the organization owners can do that.
